spec/controllers/course/leaderboards_controller_spec.rb

```diff
@@ -11,12 +11,25 @@
     before { sign_in(user) }
 
     describe '#show' do
-      before do
-        allow(controller).to receive_message_chain('current_component_host.[]').and_return(nil)
-      end
       subject { get :show, course_id: course }
-      it 'raises an component not found error' do
-        expect { subject }.to raise_error(ComponentNotFoundError)
+
+      context 'when the leaderboard component is disabled' do
+        before do
+          allow(controller).to receive_message_chain('current_component_host.[]').and_return(nil)
+        end
+        it 'raises an component not found error' do
+          expect { subject }.to raise_error(ComponentNotFoundError)
+        end
+      end
+    end
+
+    describe '#groups' do
+      subject { get :groups, course_id: course }
+
+      context' when group leaderboards are disabled in leaderboard settings' do
```

Comment:
shall we name the group leaderboard in a singular form: "group leaderboard is disabled"  ?